### PR TITLE
Refractor Notebooks

### PR DESCRIPTION
**Pull request by kabu00002**
## Description
Refractor notebooks: remove out-dated notebooks, add structure and explanations to notebooks, and outsource functionality

## Todos
  - [x] Refractor diversity notebook
  - [x] Refractor chemical  properties notebook
  - [x] Refractor novelty notebook
  - [x] Make `src` as package available and outsource notebook functions to `src/utils/utils_eval`

## Status
- [x] Ready to go

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ channels:
   - conda-forge
 dependencies:
   # Base dependencies
-  - python=3.8
+  - python>=3.8
```
[the remark this answers is not in the document]
Comment (kabu00002) [reply environment.yml:6]: Yes

File: src/evaluation/chembl/chembl_database_comp.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    )
+    parser.add_argument(
+        "-m",
+        "--use_morgan",
+        action="store_true",
+        help="If flag is set, Morgan fingerprints will be used. Otherwise (by default), RDKit fingerprints are used.",
+    )
+    parser.add_argument(
+        "-p",
+        "--post_filtering",
```
[the remark this answers is not in the document]
Comment (kabu00002) [reply src/evaluation/chembl/chembl_database_comp.py:24]: Done:)
 dependencies:

File: src/evaluation/utils.py [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+
+    Returns
+    -------
+    float
+        Mean tanimoto similarity between fingerprint and all_fingerprints (excluding self similarity)
+    """
+    similarities = DataStructs.BulkTanimotoSimilarity(
+        target_fingerprint, all_fingerprints
+    )
+    similarities.remove(1)  # self similarity
```
[the remark this answers is not in the document]
Comment (kabu00002) [reply src/evaluation/utils.py:340]: Yeag taht should be fine

**Comment by kabu00002** [comment]
<p>That should be fragments that had different dummy atoms </p><p> </p>

---

 <a href='https://app.reviewnb.com/volkamerlab/KinFragLib_PocketEnum/pull/16/discussion/#comment-2407728581'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by kabu00002** [comment]
<p>I added them on top of the bar</p><p> </p>

---

 <a href='https://app.reviewnb.com/volkamerlab/KinFragLib_PocketEnum/pull/16/discussion/#comment-2407728623'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by kabu00002** [comment]
<p>These are the ligands that are duplicated (60 ligands). I would remove them in the postproccessing step</p>

---

 <a href='https://app.reviewnb.com/volkamerlab/KinFragLib_PocketEnum/pull/16/discussion/#comment-2407728651'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by kabu00002** [comment]
<p>(1) that was a small bug</p><p>(2) no, since we only traversed through AP-GA-FP</p>

---

 <a href='https://app.reviewnb.com/volkamerlab/KinFragLib_PocketEnum/pull/16/discussion/#comment-2407728712'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by kabu00002** [comment]
@PaulaKramer ready to merge?

**Comment by PaulaKramer** [comment]
Thanks, the notebooks look great! 
I think it would be nice if we structured the `evaluation` folder a little more. Here are some suggestions: 
- add a `data` folder with all csv files needed for the notebooks (change the path in the notebooks)
- add a separate folder with all .py files (called `utils` or something like that), since these are more helper functions for the notebooks right? 
- mention all the notebooks in the readme (the one in the evaluation folder) and shortly explain what they cover
- rename the notebooks to more expressive names (e.g. it's not entirely clear what is covered in the chembl.ipynb)

**Pull request merged**